fiftyone/core/utils.py

```diff
@@ -1467,10 +1468,14 @@ def __init__(
         )
         self._last_batch_content_size = 0
 
-    def apply_backpressure(self, bsonable_batch):
-        batch_content_size = sum(
-            len(json_util.dumps(obj)) for obj in bsonable_batch
-        )
+    def apply_backpressure(self, batch_or_size):
+        if isinstance(batch_or_size, numbers.Number):
+            batch_content_size = batch_or_size
+        else:
+            batch_content_size = sum(
+                len(json_util.dumps(obj)) for obj in batch_or_size
+            )
```

Comment:
If we allow this, we should probably just change the name of this class from `BSONSizeDynamicBatcher` to `ContentSizeDynamicBatcher`

Reply:
I made this change: https://github.com/voxel51/fiftyone/pull/4015/commits/fbccc7d49385c1628917d3d45c65db832ce8d5ad